internal/server/namespace.go

```diff
@@ -0,0 +1,88 @@
+package server
+
+import (
+	"context"
+	"encoding/base64"
+
+	"go.flipt.io/flipt/errors"
+	"go.flipt.io/flipt/internal/storage"
+	flipt "go.flipt.io/flipt/rpc/flipt"
+	"go.uber.org/zap"
+	empty "google.golang.org/protobuf/types/known/emptypb"
+)
+
+// GetNamespace gets a namespace
+func (s *Server) GetNamespace(ctx context.Context, r *flipt.GetNamespaceRequest) (*flipt.Namespace, error) {
+	s.logger.Debug("get namespace", zap.Stringer("request", r))
+	namespace, err := s.store.GetNamespace(ctx, r.Key)
+	s.logger.Debug("get namespace", zap.Stringer("response", namespace))
+	return namespace, err
+}
+
+// ListNamespaces lists all namespaces
+func (s *Server) ListNamespaces(ctx context.Context, r *flipt.ListNamespaceRequest) (*flipt.NamespaceList, error) {
+	s.logger.Debug("list namespaces", zap.Stringer("request", r))
+
+	if r.Offset < 0 {
+		r.Offset = 0
+	}
+
+	opts := []storage.QueryOption{storage.WithLimit(uint64(r.Limit))}
+
+	if r.PageToken != "" {
+		tok, err := base64.StdEncoding.DecodeString(r.PageToken)
+		if err != nil {
+			return nil, errors.ErrInvalidf("pageToken is not valid: %q", r.PageToken)
+		}
+
+		opts = append(opts, storage.WithPageToken(string(tok)))
+	} else if r.Offset >= 0 {
+		// TODO: deprecate
+		opts = append(opts, storage.WithOffset(uint64(r.Offset)))
+	}
+
+	results, err := s.store.ListNamespaces(ctx, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp flipt.NamespaceList
+
+	resp.Namespaces = append(resp.Namespaces, results.Results...)
```

Comment:
I might be missing something, but I think this should be safe and avoid a copy:
```suggestion
	resp.Namespaces = results.Results
```